### PR TITLE
BOM-1281 : Implement Tox in credentials

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Related JIRA Link : [https://openedx.atlassian.net/browse/BOM-1281](https://openedx.atlassian.net/browse/BOM-1281)

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+whitelist_externals:
+    npm
+    make
+commands =
+    npm install --unsafe-perm
+    npm install --production --no-save
+    make static
+    coverage run -m pytest --ds credentials.settings.test --durations=25
+    coverage report
+    $NODE_BIN/gulp test
```
Comment (mzulqarnain1) [inline tox.ini]: @feanil We are trying to move tests to tox, and this command we moved from makefile to tox.ini gives an error `ERROR: InvocationError for command could not find executable '$NODE_BIN/gulp'` We couldn't find any solution for this problem, @UsamaSadiq is facing same issue in course-discovery. Can you help us with a solution?

Comment (jmbowman) [reply tox.ini]: @mzulqarnain1 @UsamaSadiq We discussed some options for fixing the InvocationError in an earlier meeting, have those been tried yet?  I noticed that both this and the course-discovery ticket are marked as blocked, and I want to make sure you have what you need to make progress on them.

Comment (jmbowman) [reply tox.ini]: New suggestion on this: can we just break out the gulp and react tests into a separate target that isn't called from within tox?  Those tests are executed via JavaScript, so there's no reason to run them from inside a tox virtualenv.  Let's just separate the Python and JS testing targets for Travis, with a target for local development convenience that runs both.

File: package-lock.json [hunk 2]
```diff
@@ -9,8 +9,8 @@
       "integrity": "sha512-3T6yYiLK9WdIn0iwjLoWfrsDqgfd0byiVB3aXK7w4oaltdVzA/fB/qU3iKrJ8cy4T6ZgDA2uFP726fU0StmxUw==",
       "requires": {
         "bootstrap": "4.0.0",
-        "jquery": "3.3.1",
-        "popper.js": "1.14.6"
+        "jquery": "^3.2.1",
+        "popper.js": "^1.12.6"
```
Comment (jmbowman) [inline package-lock.json]: It looks like a lot of npm packages were downgraded here; was this necessary, or was checking in changes to this file accidental?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,22 @@
+[tox]
+envlist = {py35}-django{111,20,21,22}
+skipdist = true
```
Comment (jmbowman) [inline tox.ini]: This should be "skipsdist".  Fixing the spelling should eliminate the need for a `setup.py` file.

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    django20: Django>=2.0,<2.1
+    django21: Django>=2.1,<2.2
+    django22: Django>=2.2,<2.3
+    -r {toxinidir}/requirements/test.txt
+whitelist_externals:
+    make
+commands =
+    make js-requirements
+    make static
+    coverage run -m pytest --ds credentials.settings.test --durations=25
```
Comment (jmbowman) [inline tox.ini]: You shouldn't need to specify the settings file here since it's in the pytest configuration above.

File: package-lock.json (hunk 2 above)
Comment (mzulqarnain1) [reply package-lock.json]: I'll drop this file's changes. Not really needed.

File: tox.ini (hunk 3 above)
Comment (mzulqarnain1) [reply tox.ini]: nice catch. Thanks.

File: tox.ini (hunk 4 above)
Comment (mzulqarnain1) [reply tox.ini]: will remove.

**Pull request merged**